tests/deployment/sagemaker/test_sagemaker.py

```diff
@@ -1,33 +1,419 @@
+    sagemaker_client = boto3.client('sagemaker', 'us-west-2')
+    stubber = Stubber(sagemaker_client)
+    stubber.add_client_error(
+        method='delete_model', service_error_code='ValidationException'
+    )
+    stubber.activate()
+
+    error_status = _cleanup_sagemaker_model(
+        sagemaker_client, 'test_name', 'test_version'
+    )
+    assert error_status.status_code == Status.NOT_FOUND
+
+    stubber.add_client_error(
+        method='delete_model', service_error_code='InvalidSignatureException'
+    )
+    error_status = _cleanup_sagemaker_model(
+        sagemaker_client, 'test_name', 'test_version'
+    )
+    assert error_status.status_code == Status.UNAUTHENTICATED
+
+    stubber.add_client_error(
+        method='delete_model',
+        service_error_code='RandomError',
+        service_message='random',
+    )
+    with pytest.raises(ClientError) as error:
+        _cleanup_sagemaker_model(sagemaker_client, 'test_name', 'test_version')
+    assert error.value.operation_name == 'DeleteModel'
+    assert error.value.response['Error']['Code'] == 'RandomError'
+
+
+def test_cleanup_sagemaker_endpoint_config(sagemaker_client):
+    stubber = Stubber(sagemaker_client)
+    stubber.add_client_error(
+        method='delete_endpoint_config', service_error_code='ValidationException'
+    )
+    stubber.activate()
+
+    error_status = _cleanup_sagemaker_endpoint_config(
+        sagemaker_client, 'test_name', 'test_version'
+    )
+    assert error_status.status_code == Status.NOT_FOUND
+
+    stubber.add_client_error(
+        method='delete_endpoint_config', service_error_code='InvalidSignatureException'
+    )
+    error_status = _cleanup_sagemaker_endpoint_config(
+        sagemaker_client, 'test_name', 'test_version'
+    )
+    assert error_status.status_code == Status.UNAUTHENTICATED
+
+    stubber.add_client_error(
+        method='delete_endpoint_config',
+        service_error_code='RandomError',
+        service_message='random',
+    )
+    with pytest.raises(ClientError) as error:
+        _cleanup_sagemaker_endpoint_config(
+            sagemaker_client, 'test_name', 'test_version'
+        )
+    assert error.value.operation_name == 'DeleteEndpointConfig'
+    assert error.value.response['Error']['Code'] == 'RandomError'
+
+
+ROLE_PATH_ARN_RESULT = 'arn:aws:us-west-2:999'
+USER_PATH_ARN_RESULT = 'arn:aws:us-west-2:888'
+
+
+def test_get_arn_from_aws_user():
+    def mock_role_path_call(self, operation_name, kwarg):
+        if operation_name == 'GetCallerIdentity':
+            return {'Arn': 'something:something:role/random'}
+        elif operation_name == 'GetRole':
+            return {'Role': {'Arn': ROLE_PATH_ARN_RESULT}}
+        else:
+            raise Exception(
+                'This test does not handle operation: {}'.format(operation_name)
+            )
+
+    @patch('botocore.client.BaseClient._make_api_call', new=mock_role_path_call)
+    def role_path():
```

Comment:
use the following syntax  instead of wrapping function:

```
with patch('botocore.client.BaseClient._make_api_call', new=mock_role_path_call): 
    assert get_arn_role_from_current_aws_user() == '....'
```